### ISSUE TITLE
Enum deserialization should be case-insensitive

### ISSUE THREAD

**Issue by DougSchmidt-AI** [ServiceStack/ServiceStack.Java#29]
Just had this bug reported by a consumer of our Java SDK, which is based on `ServiceStack.Java`.

GSON needs to be configured to deserialize enumeration values in a case-insensitive manner.

Basically, the ServiceStack Java generator calls [`enumValueName.ToPascalCase()`](https://github.com/ServiceStack/ServiceStack/blob/master/src/ServiceStack/NativeTypes/Java/JavaGenerator.cs#L317) when generating the Java types. But when ServiceStack.Text serializes the values to JSON, the original enumeration value name is preserved.

If the original enumeration value name is different from the `ToPascalCase()` output, then GSON doesn't recognize the text and deserializes a `null` instead.

[Here](https://github.com/AquaticInformatics/aquarius-sdk-net/blob/develop/src/Aquarius.Client/TimeSeries/Client/ServiceModels/Publish.cs#L295) is an original C# enumeration with the value `USGSMultiPoint`.
[Here](https://github.com/AquaticInformatics/aquarius-sdk-java/blob/develop/src/main/java/com/aquaticinformatics/aquarius/sdk/timeseries/servicemodels/Publish.java#L5252) is the generated Java enumeration with the value `UsgsMultiPoint` from the `ToPascalCase()` conversion.

I worked around this bug in our SDK, with [this commit](https://github.com/AquaticInformatics/aquarius-sdk-java/commit/8747469921d4d4037ca846c2e123e265ce301778), which should probably be incorporated into ServiceStack.Java pretty much as-is.

**Comment by mythz** [comment]
> I worked around this bug in our SDK, with this commit, which should probably be incorporated into ServiceStack.Java pretty much as-is.

Hi Doug,

Can you either approve that code in [this commit](https://github.com/AquaticInformatics/aquarius-sdk-java/commit/8747469921d4d4037ca846c2e123e265ce301778) can be re-licensed as BSD so it in can be used in ServiceStack.Java, or copy the source code into a GIST (doesn't have to compile) with a BSD license header so I can copy it from that instead. Or if preferred a PR will work too.

**Comment by mythz** [comment]
You're using the [GPL Affero license](https://github.com/AquaticInformatics/aquarius-sdk-java/blob/develop/LICENSE.txt) but that prevents it from being used in closed source projects. ServiceStack uses a dual-licensing model for its Server libraries where Customers are using ServiceStack under the commercial license which allows usage in closed source projects. But for stuff like Live Demos or external client libraries we license them under a [BSD license](https://github.com/ServiceStack/ServiceStack.Java/blob/master/license.txt) where it allows usage in closed source projects and other OSS licensed projects (i.e. so it can be used by Customers without a commercial license).

Basically if you change your project to use any of [popular OSI Licenses](https://opensource.org/licenses) it will allow usage in other OSS licensed projects and closed source projects.

**Comment by schmidt4brains** [comment]
Hmm, that's not my understanding of Affero, but I totally could have gotten that wrong. My understanding was that Affero is fine for commercial use as-is. You only need to disclose the source if you modify or derive, but not consume-without-modification. But I'll look into that again (no need for this thread to turn into a FOSS bun fight!)

I also see that the ServiceStack.Java repo only had its license added after I raised this issue, so I don't feel too bad about not getting this stuff right on the first try. :)

**Comment by schmidt4brains** [comment]
Any suggestions for starting points for repo bulding?

A stock `mvn clean install` command for any folder with a `pox.xml` tends to give lots of "OMG this isn't Java 1.3" warnings.

I don't know enough about the building/targeting expectations for Android code to know if I should just add the standard `<java.version>1.8</java.version>` properties in the pom, or if something else is required.

**Comment by mythz** [comment]
> My understanding was that Affero is fine for commercial use as-is. You only need to disclose the source if you modify or derive, but not consume-without-modification.

It requires that the source code of any derivative project making use of it is made available as well, i.e. prohibiting usage in closed source projects. 

> Any suggestions for starting points for repo bulding?

It basically requires the latest version of Android Studio with latest SDK installed. My basic approach is to install Android Studio and all SDK's the projects uses, then play whack-a-mole by chasing down build errors and installing any missing components. It's the most fragile build system I've used where every update breaks the build without fail and requires manual intervention to get it building again.
